feat(Worklets): `runOnRuntimeSync` (#8224)
software-mansion/react-native-reanimated: packages/docs-worklets/docs/threading/runOnRuntimeSync.mdx

---
sidebar_position: 13
---

# runOnRuntimeSync

`runOnRuntimeSync` lets you run a [workletized](/docs/fundamentals/glossary#to-workletize) function synchronously on a [Worker Runtime](/docs/fundamentals/glossary#worker-worklet-runtime---worker-runtime).

## Reference

```javascript
import { runOnRuntimeSync, createWorkletRuntime } from 'react-native-worklets';

const workletRuntime = createWorkletRuntime({ name: 'background' });

const result = runOnRuntimeSync(workletRuntime, () => {
  'worklet';
  return 2+2;
});

console.log(result); // 4
```

<details>
<summary>Type definitions</summary>

```typescript
function runOnRuntimeSync<Args extends unknown[], ReturnValue>(
  workletRuntime: WorkletRuntime,
  worklet: (...args: Args) => ReturnValue,
  ...args: Args
): ReturnValue
```

</details>

## Arguments

### workletRuntime

The worklet runtime to run the worklet on.

### worklet

A reference to a function you want to execute on the [Worklet Runtime](/docs/fundamentals/glossary#worklet-runtime).

### args

Arguments to the function you want to execute on the [Worklet Runtime](/docs/fundamentals/glossary#worklet-runtime).

> ## Remarks

Reviewer: We should add an information about the fact that this API can effectively pre-empt the runtime from another thread, meaning it's blocking.

We could also add a little bit of information when to use `runOnRuntimeSync`/`runOnRuntimeAsync` or explain the differences.

> - The worklet cannot be run on the Worker Runtime from [UI
>   Runtime](/docs/fundamentals/glossary#ui-runtime) or another [Worker
>   Runtime](/docs/fundamentals/glossary#worker-worklet-runtime---worker-runtime), unless the [Bundle Mode](/docs/experimental/bundleMode) is enabled.

Reviewer: ```suggestion
- `runOnRuntimeSync` can only be called on the [RN Runtime](link) unless the [Bundle Mode](/docs/experimental/bundleMode) is enabled.
```
